app/commands/mentor/update_stats.rb

```diff
@@ -3,9 +3,7 @@ class Mentor::UpdateStats
 
   queue_as :default
 
-  initialize_with :mentor,
-    update_counts: false,
-    update_satisfaction_rating: false
+  initialize_with :mentor, :track
```

Comment:
We now also pass the `track` in here, which allows us to update number of finished discussions and the automator flag. This is probably the key change in the PR (or at least it was my big mental breakthrough), as it enables this command to do all the things in the correct order, without races and spagetti.